### ISSUE TITLE
[14.0.3] InfluxDB incorrect timestamp, no timezone used

### ISSUE THREAD

**Issue by dstoianov** [jomjol/AI-on-the-edge-device#1991]
### The Problem

After updating device it starts Incorrectly sending timestamps in case of direct saving the data in influxDB. 

See attached logs and screenshots 

### Version

14.0.3

### Logfile

```shell
[0d00h00m04s] 2023-02-06T18:37:11 <INF> [SNTP] TimeServer not defined, using default: pool.ntp.org
[0d00h00m04s] 2023-02-06T18:37:11 <INF> [SNTP] Configuring NTP Client...
[0d00h00m04s] 2023-02-06T19:37:11 <INF> [SNTP] Time zone set to CET-1CEST
[0d00h00m04s] 2023-02-06T19:37:11 <INF> [SNTP] Time is already set: 2023-02-06 19:37:11
[0d00h00m04s] 2023-02-06T19:37:11 <INF> [MAIN] =================================================
[0d00h00m04s] 2023-02-06T19:37:11 <INF> [MAIN] ==================== Startup ====================
[0d00h00m04s] 2023-02-06T19:37:11 <INF> [MAIN] =================================================
[0d00h00m04s] 2023-02-06T19:37:11 <INF> [MAIN] Release: v14.0.3 (Commit: 6c891ab+), Date/Time: 2023-02-05 19:36, Web UI: Release: v14.0.3 (Commit: 6c891ab+)
[0d00h00m04s] 2023-02-06T19:37:11 <INF> [MAIN] Reset reason: Via esp_restart
[0d00h00m05s] 2023-02-06T19:37:12 <INF> [MAIN] WLAN-Settings - RSSI-Threashold: 0
[0d00h00m09s] 2023-02-06T19:37:16 <INF> [MAIN] =================================================
[0d00h00m09s] 2023-02-06T19:37:16 <INF> [MAIN] ================== Main Started =================
[0d00h00m09s] 2023-02-06T19:37:16 <INF> [MAIN] =================================================
[0d00h00m09s] 2023-02-06T19:37:16 <INF> [MAIN] The device has 8 MBytes of PSRAM
[0d00h00m11s] 2023-02-06T19:37:18 <INF> [MAIN] Initialization completed successfully!
[0d00h00m16s] 2023-02-06T19:37:23 <INF> [TFLITE SERVER] Round #1 started
[0d00h00m35s] 2023-02-06T19:39:06 <INF> [SNTP] Time is now successfully synced with NTP Server pool.ntp.org: 2023-02-06 19:39:06
[0d00h01m32s] 2023-02-06T19:40:03 <INF> [POSTPROC] main: Raw: 02664.892, Value: 2664.892, Status: no error
[0d00h01m32s] 2023-02-06T19:40:03 <INF> [INFLUXDB] sending line to influxdb:gas main/value=2664.892 1675701452000000000
[0d00h01m32s] 2023-02-06T19:40:03 <INF> [INFLUXDB] API URI: http://192.168.1.11:8086/api/v2/write?bucket=sensors/
[0d00h01m32s] 2023-02-06T19:40:03 <INF> [TFLITE SERVER] Round #1 completed (76 seconds)
```

### Expected Behavior

the timestamp that sends to the DB should use the correct timezone

### Screenshots

<img width="418" alt="image" src="https://user-images.githubusercontent.com/4168915/217074639-15e708a5-a722-48ab-9ed5-358a385bf39d.png">

<img width="1073" alt="image" src="https://user-images.githubusercontent.com/4168915/217075036-0d291876-3b61-4848-a083-2f0df7863123.png">

**Comment by jomjol** [comment]
The time should be the time of taking the image which differ up to 1-2 minutes, due to the alignment procedure. 
InfluxDB expects the timestamp in UTC time and I thought, that I convert it correctly. Would need to check, but not this week.

**Comment by dstoianov** [comment]
if the current logic is correct and no bugs, all previous versions were incorrect and always sends data incorrectly
i have 2 devices one is for `wasser` and the second one for `gas`, both have different versions

Please check the attached screenshots below

<img width="1141" alt="image" src="https://user-images.githubusercontent.com/4168915/217086534-e86aadbb-43aa-4024-9046-6520f8434cbd.png">
<img width="1141" alt="image" src="https://user-images.githubusercontent.com/4168915/217087934-150f8842-6b72-4dd7-9e98-06b1d162badd.png">

**Comment by dstoianov** [comment]
as a workaround, i set time zone to  `UTC0`

**Comment by jomjol** [comment]
I found the problem and tried to correct it. Can you test this please before I commit it the main version: 
https://github.com/jomjol/AI-on-the-edge-device/actions/runs/4127451296

**Comment by jomjol** [comment]
@dstoianov : did you test it and can give feedback?

**Comment by melvinisken** [comment]
Maybe as additional remark: the timezone values are also not correctly send via mqtt. For me, it's different hours off (one is using 13.07, the other one 13.08, both are set to timezone Europe/Berlin (didn't change it, because it fit me). And the start screen also shows weird values (the time of the last page refresh is the correct time):
![grafik](https://user-images.githubusercontent.com/10966010/218336026-3a0cae3c-3df4-4f81-bbdc-695ea16721c4.png)
![grafik](https://user-images.githubusercontent.com/10966010/218336029-cf016dfb-8990-40cd-8bb3-8efdc68f67fb.png)

**Comment by dstoianov** [comment]
It seems like it works now correctly. I used such file for update 
 - **AI-on-the-edge-device__update__InfluxDB-Fix-Timeshift_(a9c5beb).zip** 

Here are the logs down below:

```java
0d00h09m08s] 2023-02-13T23:39:19 <WRN> [OTA] Reboot in 5sec
[0d00h00m04s] 2023-02-13T22:18:33 <INF> [SNTP] TimeServer not defined, using default: pool.ntp.org
[0d00h00m04s] 2023-02-13T22:18:33 <INF> [SNTP] Configuring NTP Client...
[0d00h00m04s] 2023-02-13T23:18:33 <INF> [SNTP] Time zone set to CET-1CEST
[0d00h00m04s] 2023-02-13T23:18:33 <INF> [SNTP] Time is already set: 2023-02-13 23:18:33
[0d00h00m04s] 2023-02-13T23:18:33 <INF> [MAIN] =================================================
[0d00h00m04s] 2023-02-13T23:18:33 <INF> [MAIN] ==================== Startup ====================
[0d00h00m04s] 2023-02-13T23:18:33 <INF> [MAIN] =================================================
[0d00h00m04s] 2023-02-13T23:18:33 <INF> [MAIN] Development-Branch: InfluxDB-Fix-Timeshift (Commit: a9c5beb+), Date/Time: 2023-02-08 18:57, Web UI: Development-Branch: InfluxDB-Fix-Timeshift (Commit: a9c5beb+)
[0d00h00m04s] 2023-02-13T23:18:33 <INF> [MAIN] Reset reason: Via esp_restart
[0d00h00m05s] 2023-02-13T23:18:34 <INF> [MAIN] WLAN-Settings - RSSI-Threashold: 0
[0d00h00m09s] 2023-02-13T23:18:38 <INF> [MAIN] =================================================
[0d00h00m09s] 2023-02-13T23:18:38 <INF> [MAIN] ================== Main Started =================
[0d00h00m09s] 2023-02-13T23:18:38 <INF> [MAIN] =================================================
[0d00h00m09s] 2023-02-13T23:18:38 <INF> [MAIN] The device has 8 MBytes of PSRAM
[0d00h00m11s] 2023-02-13T23:18:40 <INF> [MAIN] Initialization completed successfully!
[0d00h00m16s] 2023-02-13T23:18:45 <INF> [TFLITE SERVER] Round #1 started
[0d00h00m36s] 2023-02-13T23:40:03 <INF> [SNTP] Time is now successfully synced with NTP Server pool.ntp.org: 2023-02-13 23:40:03
[0d00h01m31s] 2023-02-13T23:40:59 <INF> [POSTPROC] main: Raw: 02725.631, Value: 2725.631, Status: no error
[0d00h01m31s] 2023-02-13T23:40:59 <INF> [INFLUXDB] sending line to influxdb:gas main/value=2725.631 1676326734000000000
[0d00h01m31s] 2023-02-13T23:40:59 <INF> [INFLUXDB] API URI: http://192.168.1.11:8086/api/v2/write?bucket=sensors/
[0d00h01m32s] 2023-02-13T23:40:59 <INF> [TFLITE SERVER] Round #1 completed (76 seconds)
[0d00h03m16s] 2023-02-13T23:42:43 <INF> [TFLITE SERVER] Round #2 started
[0d00h04m31s] 2023-02-13T23:43:58 <INF> [POSTPROC] main: Raw: 02725.631, Value: 2725.631, Status: no error
[0d00h04m31s] 2023-02-13T23:43:58 <INF> [INFLUXDB] sending line to influxdb:gas main/value=2725.631 1676328172000000000
[0d00h04m31s] 2023-02-13T23:43:58 <INF> [INFLUXDB] API URI: http://192.168.1.11:8086/api/v2/write?bucket=sensors/
[0d00h04m31s] 2023-02-13T23:43:58 <INF> [TFLITE SERVER] Round #2 completed (75 seconds)
[0d00h06m16s] 2023-02-13T23:45:43 <INF> [TFLITE SERVER] Round #3 started
[0d00h07m31s] 2023-02-13T23:46:58 <INF> [POSTPROC] main: Raw: 02725.631, Value: 2725.631, Status: no error
[0d00h07m31s] 2023-02-13T23:46:58 <INF> [INFLUXDB] sending line to influxdb:gas main/value=2725.631 1676328352000000000
[0d00h07m31s] 2023-02-13T23:46:58 <INF> [INFLUXDB] API URI: http://192.168.1.11:8086/api/v2/write?bucket=sensors/
[0d00h07m31s] 2023-02-13T23:46:58 <INF> [TFLITE SERVER] Round #3 completed (75 seconds)
[0d00h09m16s] 2023-02-13T23:48:43 <INF> [TFLITE SERVER] Round #4 started
```

Thank you

**Comment by caco3** [comment]
contained in rolling now
